src/Cardano/CoinSelection/LargestFirst.hs

```diff
@@ -94,28 +95,34 @@ atLeast
     :: ([(TxIn, TxOut)], CoinSelection)
     -> TxOut
     -> Maybe ([(TxIn, TxOut)], CoinSelection)
-atLeast (utxo0, selection) txout =
-    coverOutput (fromIntegral $ getCoin $ coin txout, mempty) utxo0
+atLeast (utxoAvailable, currentSelection) txout =
+    let target = fromIntegral $ getCoin $ coin txout in
+    coverTarget target utxoAvailable mempty
   where
-    coverOutput
-        :: (Integer, [(TxIn, TxOut)])
+    coverTarget
+        :: Integer
+        -> [(TxIn, TxOut)]
         -> [(TxIn, TxOut)]
         -> Maybe ([(TxIn, TxOut)], CoinSelection)
-    coverOutput (target, ins) utxo
+    coverTarget target utxoRemaining utxoSelected
```

Comment:
assuming the below refactoring does not destroy anything;-)

Reply:
All the tests still pass!